pennylane/transforms/resolve_dynamic_wires.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains a transform for resolving dynamic wires into real wires.
+"""
+from collections.abc import Hashable, Sequence
+from typing import Optional
+
+from pennylane.allocation import Allocate, Deallocate
+from pennylane.measurements import measure
+from pennylane.tape import QuantumScript, QuantumScriptBatch
+from pennylane.typing import PostprocessingFn, Result, ResultBatch
+
+from .core import transform
+
+
+class _WireManager:
+
+    def __init__(self, zeroed=(), dirty=(), min_integer=None):
+        self._zeroed = list(zeroed)
+        self._dirty = list(dirty)
+        self._loaned = {}  # wire to final register type
```

Comment:
"loan" is close to "borrow" in namespace, maybe?
Would `self._in_use` or `self._occupied` make sense?

Reply:
I think "loaned" makes the most sense to me.  We had it, gave to someone else, and then expect it back later.

Reply:
I see, yeah that makes sense :)